eZ/Bundle/EzPublishCoreBundle/EventListener/PreviewRequestListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * File containing the PreviewRequestListener class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class PreviewRequestListener implements EventSubscriberInterface
+{
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            KernelEvents::REQUEST => array('onKernelRequest', 200),
+        );
+    }
+
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if ($event->getRequestType() == HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
+
+        $parentRequest = $this->requestStack->getParentRequest();
+        if ($parentRequest->get('isPreview', false)) {
```

Comment:
Could you please move `isPreview` literal to const? 

I could be wrong here but current implementation opens kinda security issue:  `\Symfony\Component\HttpFoundation\Request::get` will also return true if isPreview is available in the query string. Please check it directly against the attributes parameter bug.

Reply:
Good catch! Wasn't aware of that - thanks!